ark-cli/src/commands/backup.rs

```diff
@@ -0,0 +1,91 @@
+use std::io::Write;
+use std::path::PathBuf;
+
+use crate::{
+    create_dir_all, dir, discover_roots, home_dir, storages_exists, timestamp,
+    AppError, CopyOptions, File, ARK_BACKUPS_PATH, ARK_FOLDER,
+    ROOTS_CFG_FILENAME,
+};
+
+#[derive(Clone, Debug, clap::Args)]
+#[clap(name = "backup", about = "Backup the ark managed folder")]
+pub struct Backup {
+    #[clap(value_parser, help = "Path to the root directory")]
+    roots_cfg: Option<PathBuf>,
+}
+
+impl Backup {
+    // FIXME: The logic for backup should be moved out of `ark-cli`
+    //        so that it can be used by other applications
```

Comment:
Good idea to make a crate from it, so developers could embed backup feature into their apps.

Let's create the issue for this.

Reply:
Thanks. I have opened issue #68 to keep track of it.

Reply:
Let's remove this fixme then.